### DISCUSSION TITLE
iOS publishing fails with Xcode 16.2 Beta 3 version

### DISCUSSION THREAD

**Discussion by luis901101** [codemagic-ci-cd/codemagic-docs#2882] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Using Flutter 3.27.0 with Xcode 16.2 to build and publish iOS app to App Store Connect
<img width="1942" alt="Screenshot 2024-12-12 at 17 59 19" src="https://github.com/user-attachments/assets/cb49045d-8a65-4319-b018-ea4e824a1e8d" />

### Expected results

The app to be published successfully

### Actual results

Publish failed:

```bash
2024-12-12 17:37:26.561 *** Error: [ContentDelivery.Uploader.600000420180] Asset validation failed (90534) Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https://developer.apple.com/news/releases. (ID: bff13d2e-210f-4509-bc27-65d67a31f7d3)
2024-12-12 17:37:26.770 [ContentDelivery.Uploader.600000420180] 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"8.003.16003","tool-path":"\/Applications\/Xcode-16.2.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"15.1.1","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases., id=bff13d2e-210f-4509-bc27-65d67a31f7d3, code=STATE_ERROR.VALIDATION_ERROR.90534, title=Asset validation failed, NSLocalizedFailureReason=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90534","NSLocalizedFailureReason":"Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn’t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https:\/\/developer.apple.com\/news\/releases. (ID: bff13d2e-210f-4509-bc27-65d67a31f7d3)"},"code":90534}]}
```

### Build id (optional)

675b1d5e4ede54c8c30897ed

**Comment by luis901101** [comment]
I just retried the same build but using Xcode 16.1 and it published successfully, so my question, is the Xcode 16.2 available a Beta release or the final `Xcode 16.2 Build 16C5032a from 11 Dec 2024`

**Comment by luis901101** [comment]
Ok I just checked it and the Xcode 16.2 Build version in use is the Beta 3:

<img width="634" alt="Xcode vm Beta 3" src="https://github.com/user-attachments/assets/4b53b44e-bd4f-49fc-8cdd-6e48db18a7e4" />

That is confusing because it doesn't indicate that it's the Beta 3 Build:
<img width="1890" alt="Screenshot 2024-12-12 at 18 14 17" src="https://github.com/user-attachments/assets/89bbe27e-6279-4b14-b0dc-a1ba84d450cc" />

So please when adding Beta versions of Xcode put some tag like Xcode 16.2 (Beta 3) or even better if you include the Build number so we can know which one to select to deploy to App Store Connect.

**Comment by mikhail-tokarev** [comment, marked as answer]
Hi @luis901101 
Thank you for your feedback! We are currently finishing the process of updating all machines to the new Xcode 16.2 version. If you try again, it will use the release version already.
